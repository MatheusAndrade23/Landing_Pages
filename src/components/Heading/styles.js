import styled, { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, colorDark }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
  `}
`;
