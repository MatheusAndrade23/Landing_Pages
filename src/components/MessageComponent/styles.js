import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    position: fixed;
    background-color: ${theme.colors.white};
    padding: 0 40px;
    width: max-content;
    z-index: 10;
    border: 2px solid ${theme.colors.secondaryColor};
    border-radius: 10px;

    top: 70px;
    right: 20px;
  `}
`;

export const Text = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.primaryColor};
    font-style: bold;
    font-family: ${theme.fonts.family.secondary};
  `}
`;
