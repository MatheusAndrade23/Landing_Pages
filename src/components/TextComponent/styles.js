import styled, { css } from 'styled-components';

export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.medium};
  `}
`;
