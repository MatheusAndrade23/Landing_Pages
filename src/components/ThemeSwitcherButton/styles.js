import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme }) => css`
    position: fixed;
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 20rem;
    height: 4rem;
    bottom: 2rem;
    right: 8rem;
    z-index: 6;
    opacity: 0.7;
    cursor: pointer;
    font-weight: bold;
  `}
`;
