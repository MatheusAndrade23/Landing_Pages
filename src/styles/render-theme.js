import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { DefaultTheme } from './DefaultTheme';

export const renderTheme = (children) => {
  return render(<ThemeProvider theme={DefaultTheme}>{children}</ThemeProvider>);
};
