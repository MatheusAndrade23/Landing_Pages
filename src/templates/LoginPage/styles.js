import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};

    height: 100vh;
    width: 100%;

    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;

export const Form = styled.div`
  ${({ theme }) => css`
    border: 2px solid ${theme.colors.secondaryColor};
    border-radius: 10px;
    padding: 20px 35px 40px 35px;

    display: flex;
    flex-direction: column;
  `}
`;

export const Label = styled.label`
  ${({ theme }) => css`
    color: ${theme.colors.secondaryColor};
  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    outline: none;
    border: 1px solid ${theme.colors.secondaryColor};
    border-radius: 5px;
    padding: 5px;
    margin: 5px 0;
    font-size: 15px;
    color: ${theme.colors.secondaryColor};
  `}
`;

export const TitleLogin = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.secondaryColor};
    margin: 20px auto;
  `}
`;

export const InputSubmit = styled.input`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.secondaryColor};
    border-radius: 5px;
    color: ${theme.colors.secondaryColor};
    background-color: transparent;
    cursor: pointer;
    height: 30px;
    width: 160px;
    margin: 10px auto;
    transition: 0.5s;
    font-size: 15px;

    &:hover {
      border-color: ${theme.colors.white};
      color: ${theme.colors.white};
    }
  `}
`;

export const OtherAction = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.secondaryColor};
    width: 250px;
    height: 20px;
    margin-top: 10px;

    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;

    & > p {
      font-size: 15px;
    }

    & > button {
      border: none;
      background: transparent;
      margin-left: 5px;
      color: ${theme.colors.secondaryColor};
      cursor: pointer;
      font-style: italic;
      font-size: 15px;
      margin-top: 2px;

      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;

export const Or = styled.p`
  ${({ theme }) => css`
    height: 15px;
    margin: 0 auto;
    color: ${theme.colors.secondaryColor};
    font-style: italic;
  `}
`;
