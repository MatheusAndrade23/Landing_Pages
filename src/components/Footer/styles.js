import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`;
