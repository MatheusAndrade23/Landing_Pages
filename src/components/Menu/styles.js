import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    border-bottom: ${theme.colors.MediumGray};
    background: ${theme.colors.white};

    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      height: 100vh;

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height 100vh;
        align-items: center;
        overflow: auto;
      }

      & ${Heading} {
        padding-bottom: ${theme.spacings.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacings.xxlarge} 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css``}
`;
