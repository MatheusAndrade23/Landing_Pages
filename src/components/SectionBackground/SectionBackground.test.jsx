import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { SectionBackground } from '.';

describe('<SectionBackground />', () => {
  it('should render', () => {
    renderTheme(<SectionBackground>Children</SectionBackground>);
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });
});
