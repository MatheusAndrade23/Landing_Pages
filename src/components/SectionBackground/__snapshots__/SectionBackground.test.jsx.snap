// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render with dark background 1`] = `
.c0 {
  background: #ffff;
  color: #0a1128;
  background: #0a1128;
  color: #ffff;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with light background 1`] = `
.c0 {
  background: #ffff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;
